src/bitcoincash_validator.ts

```diff
@@ -0,0 +1,142 @@
+import BCH from "bchaddrjs-slp";
+
+import { NetTypes } from "./types/net.types";
+import {
+  AddressFormats,
+  TAddress,
+  TBaseValidator,
+  TIsValidAddress,
+  TNetType,
+} from "./types/validators.types";
+
+const DEFAULT_ADDRESS_FORMAT = AddressFormats.legacy;
+const DEFAULT_NETWORK_TYPE = NetTypes.prod;
+
+const isAValidAddress: TIsValidAddress = (address) => {
+  try {
+    BCH.decodeAddress(address);
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+// TODO refactor
+const isValidAddressFormat = (addressFormat: TAddress): boolean => {
+  let validAddressFormat = false;
+  switch (addressFormat) {
+    case "all":
+      validAddressFormat = true;
+      break;
+    case BCH.Format.Legacy:
+      validAddressFormat = true;
+      break;
+    case BCH.Format.Bitpay:
+      validAddressFormat = true;
+      break;
+    case BCH.Format.Cashaddr:
+      validAddressFormat = true;
+      break;
+    case BCH.Format.Slpaddr:
+      validAddressFormat = true;
+      break;
+  }
+
+  return validAddressFormat;
+};
+
+const isValidNetworkType = (address: TAddress, networkType: TNetType) => {
+  if (networkType === NetTypes.prod || networkType === NetTypes.testnet) {
+    if (networkType === NetTypes.prod) {
+      return BCH.isMainnetAddress(address);
+    }
+
+    return BCH.isTestnetAddress(address);
+  }
+
+  // TODO: check if this is okay? Defaults to true if not prod or testnet? Probably 'both'?
```

Comment:
so is it okay?

Reply:
That's not my comment, was already in the lib